core/src/main/java/org/elasticsearch/index/mapper/core/KeywordFieldMapper.java

```diff
@@ -128,6 +120,8 @@ public KeywordFieldMapper build(BuilderContext context) {
                 } else if (propName.equals("ignore_above")) {
                     builder.ignoreAbove(XContentMapValues.nodeIntegerValue(propNode, -1));
                     iterator.remove();
+                } else if (TypeParsers.parseNorms(builder, propName, propNode, parserContext)) {
```

Comment:
I think that would be better. IMO new code shouldn't be bogged down with backcompat behavior. It would only send the wrong message (eg seeing a deprecation warning when using norms.enabled on a text field, when it should be an error so the user looks at docs and sees "ah, it's just norms now").